common/src/types/v0/store/node.rs

```diff
@@ -13,6 +13,104 @@ use std::collections::HashMap;
 
 pub type NodeLabels = HashMap<String, String>;
 
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq)]
+pub struct CordonedState {
+    pub cordonlabels: Vec<String>,
+}
+
+impl CordonedState {
+    pub fn new(cordonlabels: Vec<String>) -> Self {
+        Self { cordonlabels }
+    }
+    pub fn add_cordon_label(mut self, label: String) -> Self {
+        self.cordonlabels.insert(0, label);
+        self
+    }
+    pub fn remove_cordon_label(mut self, label: String) -> Self {
+        if let Some(index) = self.cordonlabels.iter().position(|l| l == &label.clone()) {
+            self.cordonlabels.remove(index);
+        }
+        self
+    }
+}
+
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq)]
+pub struct DrainingState {
```

Comment:
We could have a single struct for both draining and drained right?
If need be you can always create type aliases, eg:
```rust
struct DrainLabels { ... }
type DrainState = DrainLabels;
type DrainedState = DrainLabels;
```

Reply:
Oh right, I didn't think of that, will do.

Reply:
Now done, I opted for one type for the two drain enum variants.